### PR TITLE
chore(build): Avoid prebuilding api side, instead use an esbuild plugin

### PR DESCRIPTION
**Pull request by dac09**
### What does this PR do?

**Removes FS-based prebuild process**
This changes the API side build process to "prebuild" files in memory with babel, rather than write to the `.redwood/prebuild` folder. Instead of prebuilding to file system, then transforming with esbuild again, we now use an esbuild plugin to do the same in memory - eliminating all the unnecessary read and writes to the FS.

We should see the dev server (and build process) speed up, particularly for Windows users

---
**Notes**
a. In order to use plugins, I switched to the async esbuild build function. This is likely better for perf anyway!
b. We still prebuild files for our tests. This is still handy to do!
c. Renamed `prebuildFile` -> `transformWithBabel` to remove any confusion!
d. I've verified that you can debug the api side by setting breakpoints

### CODE REVIEW

**Review by jtoar — COMMENTED** [review]
Looking good @dac09! I was trying to come up with a few more ideas around error reporting when transforming with Babel doesn't work for whatever reason, but don't have any concrete suggestions yet. But submitting this review now to get post the two comments I had

File: packages/internal/src/build/api.ts [hunk 1]
```diff
@@ -29,44 +22,39 @@ export const cleanApiBuild = () => {
...
+      )
+
+      if (transformedCode?.code) {
+        return {
+          contents: transformedCode.code,
+          loader: 'js',
+        }
+      }
+
+      throw new Error(`Could not transform file: ${args.path}`)
```
[the remark this answers is not in the document]
Comment (dac09) [reply packages/internal/src/build/api.ts:42]: I think even if we get the error to print it won't be very legible ;)

File: packages/cli/src/commands/deploy/__tests__/nftPack.test.js [hunk 2]
```diff
@@ -13,12 +13,12 @@ const FIXTURE_PATH = path.resolve(
 
 let functionDistFiles
 
-beforeAll(() => {
+beforeAll(async () => {
```
Comment (dac09) [inline packages/cli/src/commands/deploy/__tests__/nftPack.test.js:16]: I'm going to try and refactor these tests to not need a full build.

File: packages/internal/src/build/api.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jtoar) [reply packages/internal/src/build/api.ts:39]: Looks like Babel throws its own errors if there's a parsing error or something like that. The types/docs didn't make it super clear, but I'm not worried here anymore

**Comment by dac09** [comment]
**Todo**

- [ ] Look into whether builds are ok on Windows
- [ ] Refactor nftPack tests to avoid actually building the API side

**Pull request merged**